base/inference.jl

```diff
@@ -3916,16 +3908,27 @@ function inlineable(f::ANY, ft::ANY, e::Expr, atypes::Vector{Any}, sv::Inference
         rettype = linfo.rettype
     end
 
+    inferred === nothing && return invoke_NF(argexprs0, e.typ, atypes, sv,
+                                        atype_unlimited, invoke_data)
+
     # check that the code is inlineable
-    isa(src, CodeInfo) || return invoke_NF(argexprs0, e.typ, atypes, sv,
-                                           atype_unlimited, invoke_data)
-    src = src::CodeInfo
-    ast = src.code
+    if isa(inferred, CodeInfo)
+        src = inferred
+    else
+        src = ccall(:jl_uncompress_ast, Any, (Any, Any, Cint), method, inferred, false)::CodeInfo
+    end
     if !src.inferred || !src.inlineable
         return invoke_NF(argexprs0, e.typ, atypes, sv, atype_unlimited,
                          invoke_data)
     end
 
+    if isa(inferred, CodeInfo)
+        ast = copy_exprargs(inferred.code)
+    else
+        ast = (ccall(:jl_uncompress_ast, Any, (Any, Any, Cint), method, inferred, true)::CodeInfo).code
+    end
+    ast = ast::Array{Any,1}
```

Comment:
I think this might be what's causing the issue with "`Array` has no field `code`."